python/lsst/daf/butler/tests/_testRepo.py

```diff
@@ -215,7 +221,11 @@ def _fillAllKeys(dimension: Dimension, value: Any) -> Mapping[str, Any]:
     expandedValue = {}
     for key in dimension.uniqueKeys:
         if key.nbytes:
-            castType = bytes
+            # bytes(str(...)) would normally fail, bringing us to the
+            # except clause below, and that in particular is a problem because
+            # of the skymap dimensions' bytes 'hash' field, which has a
+            # unique constraint.
+            castType = lambda x: str(x).encode()  # noqa: E731
```

Comment:
Without this, the `_findAllKeys` logic for `bytes` fields always falls through to the `except` clause below, because `bytes(int)` `bytes(str)` both raise, and hence the value is always just `b''`.  When there is a unique constraint on a bytes field - as there is for the `skymap` dimension, and when more than one record in inserted (which is more likely now that dimension values are sometimes invented as needed, and one of the tests now does it), we get a constraint violation.

I think you're right that the fallback no longer works with this, though; I'll see about fixing that.

Reply:
Got it, so it's a bug in my code. However, can you reword the comment to explicitly say that this is, logically, a conversion to `bytes`? That won't be clear once the diff is gone.